cpp/code/flight.cc

```diff
@@ -173,7 +177,41 @@ class ParquetStorageService : public arrow::flight::FlightServerBase {
   std::shared_ptr<arrow::fs::FileSystem> root_;
 };  // end ParquetStorageService
 
-arrow::Status TestPutGetDelete() {
+class HelloWorldServiceImpl : public HelloWorldService::Service {
+  grpc::Status SayHello([[maybe_unused]] grpc::ServerContext* ctx,
+                        const HelloRequest* request, HelloResponse* reply) override {
+    const std::string& name = request->name();
+    if (name.empty()) {
+      return grpc::Status(grpc::StatusCode::INVALID_ARGUMENT, "Must provide a name!");
+    }
+    reply->set_reply("Hello, " + name);
+    return grpc::Status::OK;
+  }
+};  // end HelloWorldServiceImpl
+
+class HelloWorldClient {
+ public:
+  HelloWorldClient(std::shared_ptr<grpc::Channel> channel)
+      : stub_(HelloWorldService::NewStub(channel)) {}
+
+  arrow::Result<std::string> HelloWorld(std::string name) {
+    grpc::ClientContext context;
+    HelloRequest request;
+    request.set_name(name);
+    HelloResponse response;
+    grpc::Status status = stub_->SayHello(&context, request, &response);
+    if (!status.ok()) {
+      return arrow::Status::IOError(status.error_message());
+    }
+    return response.reply();
+  }
+
+  protected:
+    std::unique_ptr<HelloWorldService::Stub> stub_;
+};  // end HelloWorldClient
+
+arrow::Status
+TestPutGetDelete() {
```

Comment:
Actually - it seems like a different formatter got used here? clang-format wouldn't do this